crates/ress/provider/src/lib.rs

```diff
@@ -0,0 +1,231 @@
+//! Reth implementation of [`reth_ress_protocol::RessProtocolProvider`].
+
+use alloy_consensus::BlockHeader as _;
+use alloy_primitives::{Bytes, B256};
+use parking_lot::Mutex;
+use reth_chain_state::{ExecutedBlock, ExecutedBlockWithTrieUpdates, MemoryOverlayStateProvider};
+use reth_evm::execute::{BlockExecutorProvider, Executor};
+use reth_primitives::{Block, BlockBody, EthPrimitives, Header, RecoveredBlock};
+use reth_primitives_traits::Block as _;
+use reth_provider::{
+    providers::{BlockchainProvider, ProviderNodeTypes},
+    BlockReader, BlockSource, ProviderError, ProviderResult, StateProvider, StateProviderFactory,
+};
+use reth_ress_protocol::RessProtocolProvider;
+use reth_revm::{database::StateProviderDatabase, db::State, witness::ExecutionWitnessRecord};
+use reth_tasks::pool::BlockingTaskPool;
+use reth_trie::{MultiProofTargets, Nibbles, TrieInput};
+use schnellru::{ByLength, LruMap};
+use std::{sync::Arc, time::Instant};
+use tracing::*;
+
+mod recorder;
+use recorder::StateWitnessRecorderDatabase;
+
+mod pending_state;
+pub use pending_state::*;
+
+/// Reth provider implementing [`RessProtocolProvider`].
+#[allow(missing_debug_implementations)]
+pub struct RethRessProtocolProvider<N: ProviderNodeTypes, E> {
+    provider: BlockchainProvider<N>,
```

Comment:
those this actually need Blockchainprovider or do we only need certain traits here?